tests/pthread/test_pthread_reltime.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright 2019 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+
+#include <iostream>
+#include <pthread.h>
+#include <emscripten.h>
+
+typedef std::chrono::milliseconds::rep TimePoint;
+
+TimePoint now() {
+    return std::chrono::duration_cast<std::chrono::milliseconds>(
+        std::chrono::steady_clock::now().time_since_epoch()).count();
+}
```

Comment:
I think i'd find this test more readable and use fewer lines of code is it was just C and used `clock_gettime`.. but thats probably personal preference so I won't ask you to change that.